src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx

```diff
@@ -529,4 +529,21 @@
   <data name="PreferIsKindFix" xml:space="preserve">
     <value>Use 'IsKind' instead of 'Kind'</value>
   </data>
+  <data name="MissingSourceGeneratorAttributeDescription" xml:space="preserve">
+    <value>'{0}' implements 'ISourceGenerator' but does not have the 'Generator' attribute.</value>
+  </data>
+  <data name="MissingSourceGeneratorAttributeMessage" xml:space="preserve">
+    <value>Missing 'Generator' attribute</value>
+  </data>
+  <data name="MissingSourceGeneratorAttributeTitle" xml:space="preserve">
+    <value>Missing 'Generator' Attribute</value>
+  </data>
+  <data name="AddGeneratorAttribute_1" xml:space="preserve">
+    <value>Apply Generator attribute for '{0}'.</value>
+    <comment>"Generator" is a named type and should not be changed</comment>
+  </data>
+  <data name="AddGeneratorAttribute_2" xml:space="preserve">
+    <value>Apply Generator attribute for both '{0}' and '{1}'.</value>
```

Comment:
The placeholders here are always going to be the same. Do we need them?

Reply:
Technically no, but this follows the previous pattern so I kept it. Seems easier to prevent typos in language names too